### PR TITLE
Add support to the faceting setting customization at the index level

### LINKED ISSUE #412: Add support to the facet setting customization at the index level
⚠️ This issue is generated, which means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Related to:
- parent issue: https://github.com/meilisearch/integration-guides/issues/205
- engine issue: https://github.com/meilisearch/specifications/pull/157, https://github.com/meilisearch/meilisearch/issues/2368

----

Add a `faceting` index settings resource to manage customization of the maximum value per facet.

- Expose `faceting` object on `/indexes/:index_uid/settings` endpoints.
- Add `GET/PATCH/DELETE` - `/indexes/:index_uid/settings/faceting` endpoints.

The `faceting` object definition is made of the following properties:
- `maxValuesPerFacet` type (`int`)

More information about what each field does could be [checked in the spec.](https://github.com/meilisearch/specifications/pull/117)

Check the other customization methods for reference like: `synonyms`, `searchableAttributes`, `filterableAttributes`.

TODO:
- [ ] Add the new methods to customize `faceting`
  - [ ] GET `/indexes/:index_uid/settings/faceting`
  - [ ] PATCH `/indexes/:index_uid/settings/faceting`
  - [ ] DELETE `/indexes/:index_uid/settings/faceting`
- [ ] Add tests

### PR DESCRIPTION
**Pull request by MaAnCoSa**
# Pull Request

## Related issue
Fixes #412 

## What does this PR do?
* Added new methods for faceting customization:
-GET /indexes/:index_uid/settings/faceting
-PATCH /indexes/:index_uid/settings/faceting
-DELETE /indexes/:index_uid/settings/faceting
* Added tests

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: src/main/java/com/meilisearch/sdk/SettingsHandler.java [hunk 1]
```diff
@@ -475,6 +476,44 @@ TaskInfo resetPaginationSettings(String uid) throws MeilisearchException {
...
+     */
+    TaskInfo updateFacetingSettings(String uid, Faceting faceting) throws MeilisearchException {
+        return httpClient.patch(
+                settingsPath(uid).addSubroute("faceting").getURL(),
+                faceting == null ? httpClient.jsonHandler.encode(faceting) : faceting,
+                TaskInfo.class);
+    }
+
+    /**
+     * Deletes the faceting settings of the index
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/SettingsHandler.java:506]:
```suggestion
     * Reset the faceting settings of the index
```
It resets this parameter to its default value of 100

**Review by alallema — COMMENTED** [review]
Thanks, @MaAnCoSa for this PR ❤️ 
The additions of code are correct but it would be necessary to remove the added files which should not be there all in the `meili_data` directory

**Comment by MaAnCoSa** [comment]
Hi @alallema! Thank you! I removed the meili_data directory and commited the change about the reset. I Hope everything is alright now.

**Review by alallema — APPROVED** [review]
Thanks @MaAnCoSa for this PR ❤️ 
LGTM! 🚀

**Comment by alallema** [comment]
bors merge

**Pull request merged**